source/common/json/json_streamer.h

```diff
@@ -1,25 +1,119 @@
 #pragma once
 
+#include <cmath>
 #include <memory>
 #include <stack>
 #include <string>
 
 #include "envoy/buffer/buffer.h"
 
+#include "source/common/buffer/buffer_util.h"
+#include "source/common/json/json_sanitizer.h"
+
 #include "absl/strings/string_view.h"
 #include "absl/types/variant.h"
 
 namespace Envoy {
 namespace Json {
 
+class Constants {
+public:
+  // Constants for common JSON values.
+  static constexpr absl::string_view True = R"(true)";
+  static constexpr absl::string_view False = R"(false)";
+  static constexpr absl::string_view Null = R"(null)";
+
+  // Constants for JSON delimiters.
+  static constexpr absl::string_view MapBeg = R"({)";
+  static constexpr absl::string_view MapEnd = R"(})";
+  static constexpr absl::string_view ArrayBeg = R"([)";
+  static constexpr absl::string_view ArrayEnd = R"(])";
+  static constexpr absl::string_view Quote = R"(")";
+  static constexpr absl::string_view Comma = R"(,)";
+};
+
+/**
+ * Helper class that provides low level methods to sanitize keys, values, and
+ * delimiters to output buffer or string.
+ * NOTE: Streamer provides high level methods based on the Serializer to construct
+ * complete JSON output. Please use Streamer as priority.
+ * NOTE: If Serializer is preferred please ensure YOU KNOW WHAT YOU ARE DOING.
+ */
+template <class OutputBuffer> class Serializer {
+public:
+  // Constructor with output buffer to write the JSON pieces.
+  Serializer(OutputBuffer& output_buffer) : output_buffer_(output_buffer) {}
+
+  /**
+   * Add raw JSON piece to the output buffer. The value could be delimiters,
+   * or sanitized JSON piece.
+   *
+   * @param value The raw JSON piece to be added.
+   */
+  void addPiece(absl::string_view value) { output_buffer_.addFragments({value}); }
```

Comment:
Like the original `addConstantString`. But it possible for users to add raw JSON piece to the buffer by this method, not just `constant` string. So, I changed the name.

Reply:
suggest calling it addWIthoutSanitizing or something like that to reflect that you are bypassing the sanitizer -- that's the main reason to use it, right?

Reply:
or addStringWithoutSanitizing.

Reply:
Hah, I just see this comment. Because when we adding a string, we will surround it with quote. So, the method is not just without sanitizing, but do nothing additionally.
So, how about to name it `addDirectly` means that will put a string to buffer directly  and no any other operations.